### PR TITLE
Bring your own model

### PR DESCRIPTION
**Pull request by George3d6**
Added the ability to add a custom model by extending the `ModelInterface` and uploading said class in an archive containing a `model.py` file.

### CODE REVIEW

**Comment by mindsdbadmin** [comment]
****CLA Assistant Lite bot**** All contributors have signed the CLA  ✍️

File: mindsdb/interfaces/custom/custom_models.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+        return models
+
+    def delete_model(self, name):
+        shutil.rmtree(self._dir(name))
+        self.dbw.unregister_predictor(name)
+
+    def rename_model(self, name, new_name):
+        shutil.move(self._dir(name), self._dir(new_name))
+        shutil.move(os.path.join(self._dir(new_name) + f'{name}.py'), os.path.join(self._dir(new_name) ,f'{new_name}.py'))
```
Comment (StpMax) [inline mindsdb/interfaces/custom/custom_models.py:138]: here probable need add register/unregister_predictor

File: mindsdb/api/mysql/mysql_proxy/mysql_proxy.py [hunk 2]
```diff
@@ -388,7 +390,10 @@ def insert_predictor_answer(self, insert):
                     default_store.delete_datasource(ds_name)
                 raise Exception(f"Column '{col}' not exists")
 
-        mdb.learn(insert['name'], ds, insert['predict'], kwargs)
+        if insert['name'] in [x['name'] for x in custom_models.get_models()]:
+            custom_models.learn(insert['name'], ds, insert['predict'], kwargs)
+        else:
+            mdb.learn(insert['name'], ds, insert['predict'], kwargs)
```
Comment (StpMax) [inline mindsdb/api/mysql/mysql_proxy/mysql_proxy.py:396]: I don't like this not obvious behavior. Probably in future we should add 'update' statement for such cases.

File: mindsdb/api/http/initialize.py [hunk 3]
```diff
@@ -12,6 +12,8 @@
 from mindsdb.__about__ import __version__ as mindsdb_version
 from mindsdb.interfaces.datastore.datastore import DataStore
 from mindsdb.interfaces.native.mindsdb import MindsdbNative
+from mindsdb.interfaces.custom.custom_models import CustomModels
+from mindsdb.utilities.config import Config
```
Comment (StpMax) [inline mindsdb/api/http/initialize.py:16]: unused import

File: mindsdb/api/http/namespaces/predictor.py [hunk 4]
```diff
@@ -146,6 +171,27 @@ def put(self, name):
 
         return '', 200
 
+@ns_conf.route('/<name>/learn')
+@ns_conf.param('name', 'The predictor identifier')
+class PredictorColumns(Resource):
```
Comment (StpMax) [inline mindsdb/api/http/namespaces/predictor.py:176]: This class name already exist

File: mindsdb/integrations/clickhouse/clickhouse.py [hunk 5]
```diff
@@ -109,6 +107,7 @@ def register_predictors(self, model_data_arr):
             columns_sql += ',`external_datasource` Nullable(String)'
             for col in model_meta['predict']:
                 columns_sql += f',`{col}_confidence` Nullable(Float64)'
+                print(model_meta)
```
Comment (StpMax) [inline mindsdb/integrations/clickhouse/clickhouse.py:110]: unnecessary print

**Review by StpMax — CHANGES_REQUESTED** [review]
need to remove `my_model.zip` and probably `test_custom_model/model.py`

**Pull request merged**